src/domain/relayer/midnight/midnight_relayer.rs

```diff
@@ -0,0 +1,975 @@
+
+        let balance_u128 = balance
+            .to_string()
+            .parse::<u128>()
+            .map_err(|e| RelayerError::ProviderError(format!("Failed to parse balance: {}", e)))?;
+
+        Ok(BalanceResponse {
+            balance: balance_u128,
+            unit: MIDNIGHT_SMALLEST_UNIT_NAME.to_string(),
+        })
+    }
+
+    async fn get_status(&self) -> Result<RelayerStatus, RelayerError> {
+        let relayer_model = &self.relayer;
+
+        let nonce = self
+            .provider
+            .get_nonce(&relayer_model.address)
+            .await
+            .map_err(|e| {
+                RelayerError::ProviderError(format!("Failed to get account details: {}", e))
+            })?;
+
+        let nonce_str = nonce.to_string();
+
+        let balance_response = self.get_balance().await?;
+
+        let pending_statuses = [TransactionStatus::Pending, TransactionStatus::Submitted];
+        let pending_transactions = self
+            .transaction_repository
+            .find_by_status(&relayer_model.id, &pending_statuses[..])
+            .await
+            .map_err(RelayerError::from)?;
+        let pending_transactions_count = pending_transactions.len() as u64;
+
+        let confirmed_statuses = [TransactionStatus::Confirmed];
+        let confirmed_transactions = self
+            .transaction_repository
+            .find_by_status(&relayer_model.id, &confirmed_statuses[..])
+            .await
+            .map_err(RelayerError::from)?;
+
+        let last_confirmed_transaction_timestamp = confirmed_transactions
+            .iter()
+            .filter_map(|tx| tx.confirmed_at.as_ref())
+            .max()
+            .cloned();
+
+        Ok(RelayerStatus::Midnight {
+            balance: balance_response.balance.to_string(),
+            pending_transactions_count,
+            last_confirmed_transaction_timestamp,
+            system_disabled: relayer_model.system_disabled,
+            paused: relayer_model.paused,
+            nonce: nonce_str,
+        })
+    }
+
+    async fn delete_pending_transactions(
+        &self,
+    ) -> Result<DeletePendingTransactionsResponse, RelayerError> {
+        println!("Midnight delete_pending_transactions...");
+        Ok(DeletePendingTransactionsResponse {
+            queued_for_cancellation_transaction_ids: vec![],
+            failed_to_queue_transaction_ids: vec![],
+            total_processed: 0,
+        })
+    }
+
+    async fn sign_data(&self, _request: SignDataRequest) -> Result<SignDataResponse, RelayerError> {
+        Err(RelayerError::NotSupported(
+            "Signing data not supported for Midnight".to_string(),
+        ))
+    }
+
+    async fn sign_typed_data(
+        &self,
+        _request: SignTypedDataRequest,
+    ) -> Result<SignDataResponse, RelayerError> {
+        Err(RelayerError::NotSupported(
+            "Signing typed data not supported for Midnight".to_string(),
+        ))
+    }
+
+    async fn rpc(
```

Comment:
Is this supported?
Maybe to throw error like above

Reply:
Good catch. I followed same pattern as Stellar for this as well :) I will throw error instead!